### PR TITLE
[NDM][Cisco SD-WAN] Do not update pagination params in-place

### PR DESCRIPTION
**Pull request by TCheruy**
### What does this PR do?
Refactor Cisco SD-WAN pagination logic to not update request params but recreate it for each next page
- This helps/fix a bug where we could pass some previous page params to the next one, leading to a failure to get the next page
- Also updates index-based pagination to start from `endId` instead of `endId+1`, as per Cisco's recomendation

### CODE REVIEW

File: pkg/collector/corechecks/network-devices/cisco-sdwan/client/request.go [hunk 1]
```diff
@@ -120,14 +119,14 @@ func getMoreEntries[T Content](client *Client, endpoint string, params map[strin
 
 		log.Tracef("Getting page %d from endpoint %s", page+1+1, endpoint)
 		// Update the params to get next API page
-		err := updatePaginationParams(currentPageInfo, params)
+		nextParams, err := getNextPaginationParams(currentPageInfo, client.maxCount)
 		if err != nil {
 			return nil, err
 		}
 		log.Tracef("Pagination params for page %d from endpoint %s : %v", page+1+1, endpoint, params)
```
Comment (FlorianVeaux) [inline pkg/collector/corechecks/network-devices/cisco-sdwan/client/request.go:126]: Should we log `nextParams` instead?

Comment (FlorianVeaux) [reply pkg/collector/corechecks/network-devices/cisco-sdwan/client/request.go:126]: Also it doesn't seem like we use `params` anywhere, should we remove this parameter from the function arguments? (and to any code that calls it)

File: pkg/collector/corechecks/network-devices/cisco-sdwan/client/request.go [hunk 2]
```diff
@@ -120,14 +119,14 @@ func getMoreEntries[T Content](client *Client, endpoint string, params map[strin
 
 		log.Tracef("Getting page %d from endpoint %s", page+1+1, endpoint)
 		// Update the params to get next API page
```
Comment (FlorianVeaux) [inline pkg/collector/corechecks/network-devices/cisco-sdwan/client/request.go:121]:
```suggestion
		// Build pagination parameters for the next API page
```

File: pkg/collector/corechecks/network-devices/cisco-sdwan/client/request.go (hunk 1 above)
Comment (TCheruy) [reply pkg/collector/corechecks/network-devices/cisco-sdwan/client/request.go]: 🤦‍♂️🤦‍♂️🤦‍♂️ good catch thanks!

**Review by FlorianVeaux — APPROVED** [review]

**Pull request merged**